src/main/java/seedu/jarvis/storage/history/JsonSerializableHistoryManager.java

```diff
@@ -0,0 +1,128 @@
+package seedu.jarvis.storage.history;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonRootName;
+
+import seedu.jarvis.commons.exceptions.IllegalValueException;
+import seedu.jarvis.logic.commands.Command;
+import seedu.jarvis.logic.commands.address.AddAddressCommand;
+import seedu.jarvis.logic.commands.address.ClearAddressCommand;
+import seedu.jarvis.logic.commands.address.DeleteAddressCommand;
+import seedu.jarvis.logic.commands.address.EditAddressCommand;
+import seedu.jarvis.model.history.HistoryManager;
+import seedu.jarvis.storage.history.commands.JsonAdaptedCommand;
+import seedu.jarvis.storage.history.commands.address.JsonAdaptedAddAddressCommand;
+import seedu.jarvis.storage.history.commands.address.JsonAdaptedClearAddressCommand;
+import seedu.jarvis.storage.history.commands.address.JsonAdaptedDeleteAddressCommand;
+import seedu.jarvis.storage.history.commands.address.JsonAdaptedEditAddressCommand;
+import seedu.jarvis.storage.history.commands.exceptions.InvalidCommandToJsonException;
+
+/**
+ * A {@code HistoryManager} that is serializable to JSON format.
+ */
+@JsonRootName(value = "historymanager")
+public class JsonSerializableHistoryManager {
+
+    public static final String MESSAGE_ERROR_CONVERTING_HISTORY_MANAGER = "Error converting HistoryManager";
+    public static final String MESSAGE_INVALID_COMMAND = "Invalid Command";
+
+    private final List<JsonAdaptedCommand> executedCommands = new ArrayList<>();
+    private final List<JsonAdaptedCommand> inverselyExecutedCommands = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonSerializableHistoryManager} with the given executed and inversely executed commands.
+     *
+     * @param executedCommands {@code List} of executed Jackson-Friendly commands.
+     * @param inverselyExecutedCommands {@code List} of inversely executed Jackson-Friendly commands
+     */
+    @JsonCreator
+    public JsonSerializableHistoryManager(@JsonProperty("executedCommands") List<JsonAdaptedCommand> executedCommands,
```

Comment:
This implementation looks promising, perhaps we should discuss and do a minimal implementation of the storage by this weekend? My concern is that all the nesting in the json files might take some time to figure out.

Reply:
Yes, the biggest hindrance is the nested objects that the commands have to store. Perhaps we can finalize certain commands so we can start working on `jackson` friendly classes for the other features.